src/transformers/Twitch.js

```diff
@@ -0,0 +1,56 @@
+import { URL } from 'url';
+
+export const isValid = url => {
+  const { pathname } = new URL(url);
+  const pathnameParts = pathname.split('/');
+
+  if (url.match('embed') || pathnameParts[2] === 'videos') {
+    return false;
+  }
+
+  if (url.match(/collection|video|channel|channel|clip/g)) {
+    return true;
+  }
+
+  if (pathnameParts.length == 2 && pathnameParts[1]) {
+    return true;
+  }
+
+  return false;
+};
+
+export const shouldTransform = url => {
+  const { host } = new URL(url);
+
+  return (
+    [
+      'player.twitch.tv',
+      'www.twitch.tv',
+      'twitch.tv',
+      'clips.twitch.tv',
+    ].includes(host) && isValid(url)
+  );
+};
```

Comment:
I think we should go more for a way like we do in the Pinterest transformer and check for `isChannel` or `isClip` or `isCollection` or `isVideo`, so we could also re-use these functions in the `getTwitchIFrameSrc` function.

```js
export const shouldTransform = url => {
  const { host, pathname, searchParams } = new URL(url);
  const trimmedPathName = getTrimmedPathName(pathname);
  const splittedTrimmedPathName = trimmedPathName.split('/');
  const urlConfig = { host, pathname, searchParams, splittedTrimmedPathName };

  return (
    isChannel(urlConfig) ||
    isClip(urlConfig) ||
    isCollection(urlConfig) ||
    isVideo(urlConfig)
  );
};
```

I would implement all 4 methods very strictly on what you _do_ need to be valid, so we can't have use-cases that would transform without being a valid url.

```js
const isFromPlayerDomainWithParam = ({ host, searchParams }, param) =>
  host === 'player.twitch.tv' && Boolean(searchParams.get(param));
const isFromTwitchDomain = host =>
  ['twitch.tv', 'www.twitch.tv'].includes(host);

const isChannel = ({ host, searchParams, splittedTrimmedPathName }) =>
  isFromPlayerDomainWithParam({ host, searchParams }, 'channel') ||
  (isFromTwitchDomain(host) &&
    splittedTrimmedPathName.length === 1 &&
    Boolean(splittedTrimmedPathName[0]));

const isClip = ({ host, splittedTrimmedPathName }) =>
  (host === 'clips.twitch.tv' &&
    splittedTrimmedPathName.length === 1 &&
    splittedTrimmedPathName[0] !== 'embed') ||
  (isFromTwitchDomain(host) &&
    splittedTrimmedPathName.length === 3 &&
    splittedTrimmedPathName[1] === 'clip');

const isCollection = ({ host, searchParams, splittedTrimmedPathName }) =>
  isFromPlayerDomainWithParam({ host, searchParams }, 'collection') ||
  (isFromTwitchDomain(host) &&
    ((splittedTrimmedPathName.length === 2 &&
      splittedTrimmedPathName[0] === 'collections') ||
      (splittedTrimmedPathName.length === 3 &&
        splittedTrimmedPathName[1] === 'collection')));

const isVideo = ({ host, searchParams, splittedTrimmedPathName }) =>
  isFromPlayerDomainWithParam({ host, searchParams }, 'video') ||
  (isFromTwitchDomain(host) &&
    splittedTrimmedPathName.length === 2 &&
    splittedTrimmedPathName[0] === 'videos');
```

Reply:
Not 100% sure if I like what I wrote above, since I see some repetition here and there, but definitely in the right direction to make the code more readable instead of very cryptic with regexes I think 🤔🙂